sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/ConfigurationRepository.java

```diff
@@ -102,20 +111,74 @@ public void run() {
                                     createSecurityIndexIfAbsent();
                                     waitForSecurityIndexToBeAtLeastYellow();
 
-                                    ConfigHelper.uploadFile(
-                                        client,
-                                        cd + "internal_users.yml",
-                                        identityIndex,
+                                    String adminPassword = ConfigHelper.generateCommonLangPassword();
+                                    LOGGER.always().log("Generated a random admin password: {}", adminPassword);
+                                    char[] clearAdminPassword = adminPassword.toCharArray();
+                                    String hashedAdminPassword = Hasher.hash(clearAdminPassword);
+                                    User admin = new User("admin", hashedAdminPassword, null);
+                                    SecurityDynamicConfiguration<User> ium = ConfigHelper.fromYamlFile(
+                                        confFile.toString(),
                                         CType.INTERNALUSERS,
-                                        DEFAULT_CONFIG_VERSION
+                                        DEFAULT_CONFIG_VERSION,
+                                        0,
+                                        0
                                     );
+                                    if (ium.exists("admin")) {
+                                        LOGGER.warn("internal_users.yml file should not contain an entry for admin");
+                                    }
+                                    ium.putCObject("admin", admin);
+                                    final String userConfigType = CType.INTERNALUSERS.toLCString();
+                                    try {
+
+                                        final IndexRequest indexRequest = new IndexRequest(identityIndex).id(userConfigType)
+                                            .opType(DocWriteRequest.OpType.CREATE)
+                                            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                                            .source(userConfigType, XContentHelper.toXContent(ium, XContentType.JSON, false));
+                                        IndexResponse response = client.index(indexRequest).actionGet();
+                                        final String res = response.getId();
+
+                                        if (!userConfigType.equals(res)) {
+                                            throw new Exception(
+                                                "FAIL: Configuration for '"
+                                                    + userConfigType
+                                                    + "' failed for unknown reasons. Pls. consult logfile of opensearch"
```

Comment:
Nit: Do you mind removing the abbreviation? I think it makes a more polished piece. Thanks.